From b97fcfcec5caebffe0b0f70e92fb920e903a7362 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Wed, 4 Dec 2013 11:22:27 +0800
Subject: [PATCH 266/304] ENGR00290615 Wifi:enable wifi on KK4.4

Change to use android private socket instead of UNIX standard socket.
For AOSP only update to use single connection to transfer commands and
events based on android private socket.
Why not align the wpa_supplicant_overlay.conf here although we don't have this
feature supported in driver?

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 common/wifi/Android.mk                  | 48 +++++++++++++++++++++++++++++++++
 common/wifi/p2p_supplicant_overlay.conf |  1 +
 common/wifi/wpa_supplicant_overlay.conf |  1 +
 evk_6sl/init.rc                         | 26 +++++++++++++-----
 imx6/etc/init.rc                        |  7 +----
 imx6/imx6.mk                            |  3 ++-
 sabreauto_6q/init.rc                    | 24 ++++++++++++-----
 sabresd_6dq/init.rc                     | 26 +++++++++++++-----
 8 files changed, 109 insertions(+), 27 deletions(-)
 create mode 100644 common/wifi/Android.mk
 create mode 100644 common/wifi/p2p_supplicant_overlay.conf
 create mode 100644 common/wifi/wpa_supplicant_overlay.conf

diff --git a/common/wifi/Android.mk b/common/wifi/Android.mk
new file mode 100644
index 0000000..8ebbfdc
--- /dev/null
+++ b/common/wifi/Android.mk
@@ -0,0 +1,48 @@
+#
+# Copyright 2012 The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This contains the module build definitions for the hardware-specific
+# components for this device.
+#
+# As much as possible, those components should be built unconditionally,
+# with device-specific names to avoid collisions, to avoid device-specific
+# bitrot and build breakages. Building a component unconditionally does
+# *not* include it on all devices, so it is safe even with hardware-specific
+# components.
+
+ifneq ($(filter sabresd_6dq sabreauto_6q evk_6sl arm2_6sl arm2_6dq,$(TARGET_DEVICE)),)
+
+LOCAL_PATH := $(call my-dir)
+
+include $(CLEAR_VARS)
+LOCAL_MODULE       := wpa_supplicant_overlay.conf
+LOCAL_MODULE_TAGS  := optional
+LOCAL_MODULE_CLASS := ETC
+LOCAL_SRC_FILES    := $(LOCAL_MODULE)
+LOCAL_MODULE_PATH  := $(TARGET_OUT_ETC)/wifi
+include $(BUILD_PREBUILT)
+
+include $(CLEAR_VARS)
+LOCAL_MODULE       := p2p_supplicant_overlay.conf
+LOCAL_MODULE_TAGS  := optional
+LOCAL_MODULE_CLASS := ETC
+LOCAL_SRC_FILES    := $(LOCAL_MODULE)
+LOCAL_MODULE_PATH  := $(TARGET_OUT_ETC)/wifi
+include $(BUILD_PREBUILT)
+
+endif
+
+include $(call first-makefiles-under,$(LOCAL_PATH))
diff --git a/common/wifi/p2p_supplicant_overlay.conf b/common/wifi/p2p_supplicant_overlay.conf
new file mode 100644
index 0000000..acbace2
--- /dev/null
+++ b/common/wifi/p2p_supplicant_overlay.conf
@@ -0,0 +1 @@
+disable_scan_offload=1
diff --git a/common/wifi/wpa_supplicant_overlay.conf b/common/wifi/wpa_supplicant_overlay.conf
new file mode 100644
index 0000000..acbace2
--- /dev/null
+++ b/common/wifi/wpa_supplicant_overlay.conf
@@ -0,0 +1 @@
+disable_scan_offload=1
diff --git a/evk_6sl/init.rc b/evk_6sl/init.rc
index b74d9fb..9a63de1 100644
--- a/evk_6sl/init.rc
+++ b/evk_6sl/init.rc
@@ -99,22 +99,34 @@ on boot
     mount debugfs none /sys/kernel/debug/
 
 service p2p_supplicant /system/bin/wpa_supplicant \
-    -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf -N \
-    -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf
+    -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf \
+    -I/system/etc/wifi/p2p_supplicant_overlay.conf -N \
+    -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf \
+    -I/system/etc/wifi/wpa_supplicant_overlay.conf \
+    -O/data/misc/wifi/sockets -puse_p2p_group_interface=1 \
+    -e/data/misc/wifi/entropy.bin -g@android:wpa_wlan0
     class late_start
+    socket wpa_wlan0 dgram 660 wifi wifi
     disabled
     oneshot
 
-service rtw_suppl_con /system/bin/rtl_wpa_supplicant \
-    -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf -e/data/misc/wifi/entropy.bin -N \
-    -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf
+service rtw_suppl_con /system/bin/rtl_wpa_supplicant -dddd \
+    -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf \
+    -I/system/etc/wifi/p2p_supplicant_overlay.conf -N \
+    -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf \
+    -I/system/etc/wifi/wpa_supplicant_overlay.conf \
+    -O/data/misc/wifi/sockets -puse_p2p_group_interface=1 \
+    -e/data/misc/wifi/entropy.bin -g@android:wpa_wlan0
     class late_start
     socket wpa_wlan0 dgram 660 wifi wifi
     disabled
     oneshot
 
-service wpa_supplicant /system/bin/wpa_supplicant -iwlan0 -Dnl80211
-    -c/data/misc/wifi/wpa_supplicant.conf
+service wpa_supplicant /system/bin/wpa_supplicant \
+    -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf \
+    -I/system/etc/wifi/wpa_supplicant_overlay.conf \
+    -O/data/misc/wifi/sockets \
+    -e/data/misc/wifi/entropy.bin -g@android:wpa_wlan0
     socket wpa_wlan0 dgram 660 wifi wifi
     class late_start
     disabled
diff --git a/imx6/etc/init.rc b/imx6/etc/init.rc
index 654836b..14274a3 100755
--- a/imx6/etc/init.rc
+++ b/imx6/etc/init.rc
@@ -235,11 +235,10 @@ on post-fs-data
     mkdir /data/misc/systemkeys 0700 system system
     # give system access to wpa_supplicant.conf for backup and restore
     mkdir /data/misc/wifi 0770 wifi wifi
-    chmod 0770 /data/misc/wifi
     mkdir /data/misc/wifi/sockets 0770 wifi wifi
     mkdir /data/misc/wifi/wpa_supplicant 0770 wifi wifi
     mkdir /data/misc/dhcp 0770 dhcp dhcp
-    chown dhcp dhcp /data/misc/dhcp
+    chown dhcp system /data/misc/dhcp
     chmod 0660 /data/misc/wifi/wpa_supplicant.conf
     chmod 0775 /data/misc/wifi/ipconfig.txt
     chmod 0775 /system/etc/dhcpcd/dhcpcd-run-hooks
@@ -258,10 +257,6 @@ on post-fs-data
     mkdir /data/property 0700 root root
     mkdir /data/ssh 0750 root shell
     mkdir /data/ssh/empty 0700 root root
-    mkdir /data/system 0775 system system
-    mkdir /data/system/wpa_supplicant 0771 wifi wifi
-    chmod 0771 /data/system/wpa_supplicant
-    symlink /data/misc/wifi/wpa_supplicant /data/system/wpa_supplicant
 
     # create dalvik-cache, so as to enforce our permissions
     mkdir /data/dalvik-cache 0771 system system
diff --git a/imx6/imx6.mk b/imx6/imx6.mk
index 1ada99e..555bae9 100644
--- a/imx6/imx6.mk
+++ b/imx6/imx6.mk
@@ -38,7 +38,8 @@ PRODUCT_PACKAGES += \
 	wpa_supplicant				\
 	rtl_wpa_supplicant			\
 	wpa_supplicant.conf			\
-	wpa_supplicant_p2p.conf			\
+	p2p_supplicant_overlay.conf			\
+	wpa_supplicant_overlay.conf			\
 	dispd					\
 	ts_calibrator				\
 	libion                                  \
diff --git a/sabreauto_6q/init.rc b/sabreauto_6q/init.rc
index 7275483..7e4b44a 100644
--- a/sabreauto_6q/init.rc
+++ b/sabreauto_6q/init.rc
@@ -104,22 +104,34 @@ on boot
     mount debugfs none /sys/kernel/debug/
 
 service p2p_supplicant /system/bin/wpa_supplicant \
-    -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf -N \
-    -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf
+    -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf \
+    -I/system/etc/wifi/p2p_supplicant_overlay.conf -N \
+    -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf \
+    -I/system/etc/wifi/wpa_supplicant_overlay.conf \
+    -O/data/misc/wifi/sockets -puse_p2p_group_interface=1 \
+    -e/data/misc/wifi/entropy.bin -g@android:wpa_wlan0
     class late_start
+    socket wpa_wlan0 dgram 660 wifi wifi
     disabled
     oneshot
 
 service rtw_suppl_con /system/bin/rtl_wpa_supplicant \
-    -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf -e/data/misc/wifi/entropy.bin -N \
-    -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf
+    -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf \
+    -I/system/etc/wifi/p2p_supplicant_overlay.conf -N \
+    -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf \
+    -I/system/etc/wifi/wpa_supplicant_overlay.conf \
+    -O/data/misc/wifi/sockets -puse_p2p_group_interface=1 \
+    -e/data/misc/wifi/entropy.bin -g@android:wpa_wlan0
     class late_start
     socket wpa_wlan0 dgram 660 wifi wifi
     disabled
     oneshot
 
-service wpa_supplicant /system/bin/wpa_supplicant -iwlan0 -Dnl80211
-    -c/data/misc/wifi/wpa_supplicant.conf
+service wpa_supplicant /system/bin/wpa_supplicant \
+    -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf \
+    -I/system/etc/wifi/wpa_supplicant_overlay.conf \
+    -O/data/misc/wifi/sockets \
+    -e/data/misc/wifi/entropy.bin -g@android:wpa_wlan0
     socket wpa_wlan0 dgram 660 wifi wifi
     class late_start
     disabled
diff --git a/sabresd_6dq/init.rc b/sabresd_6dq/init.rc
index e8f962c..2c91838 100755
--- a/sabresd_6dq/init.rc
+++ b/sabresd_6dq/init.rc
@@ -121,22 +121,34 @@ on charger
 
 
 service p2p_supplicant /system/bin/wpa_supplicant \
-    -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf -N \
-    -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf
+    -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf \
+    -I/system/etc/wifi/p2p_supplicant_overlay.conf -N \
+    -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf \
+    -I/system/etc/wifi/wpa_supplicant_overlay.conf \
+    -O/data/misc/wifi/sockets -puse_p2p_group_interface=1 \
+    -e/data/misc/wifi/entropy.bin -g@android:wpa_wlan0
     class late_start
+    socket wpa_wlan0 dgram 660 wifi wifi
     disabled
     oneshot
 
-service rtw_suppl_con /system/bin/rtl_wpa_supplicant \
-    -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf -e/data/misc/wifi/entropy.bin -N \
-    -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf
+service rtw_suppl_con /system/bin/rtl_wpa_supplicant -dddd \
+    -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf \
+    -I/system/etc/wifi/p2p_supplicant_overlay.conf -N \
+    -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf \
+    -I/system/etc/wifi/wpa_supplicant_overlay.conf \
+    -O/data/misc/wifi/sockets -puse_p2p_group_interface=1 \
+    -e/data/misc/wifi/entropy.bin -g@android:wpa_wlan0
     class late_start
     socket wpa_wlan0 dgram 660 wifi wifi
     disabled
     oneshot
 
-service wpa_supplicant /system/bin/wpa_supplicant -iwlan0 -Dnl80211
-    -c/data/misc/wifi/wpa_supplicant.conf
+service wpa_supplicant /system/bin/wpa_supplicant \
+    -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf \
+    -I/system/etc/wifi/wpa_supplicant_overlay.conf \
+    -O/data/misc/wifi/sockets \
+    -e/data/misc/wifi/entropy.bin -g@android:wpa_wlan0
     socket wpa_wlan0 dgram 660 wifi wifi
     class late_start
     disabled
-- 
1.8.0

